h5py/h5d.pyx

```diff
@@ -430,3 +431,65 @@ cdef class DatasetID(ObjectID):
                 efree(offset)
                 if space_id:
                     H5Sclose(space_id)
+
+    IF HDF5_VERSION >= (1, 10, 2):
+
+        def read_direct_chunk(self, offsets, filter_mask=0xFFFF, PropID dxpl=None):
+            """ (offsets, H5Z_filter_t filter_mask=[], PropID dxpl=None)
+
+            Reads data to a bytes array directly from a chunk at position
+            specified by the offsets argument.
+
+            Returns a tuple containing the bytes data, and the filter_mask which
+            was used for this data.
+
+            Feature requires: 1.10.2 HDF5
+            """
+
+            cdef hid_t dset_id
+            cdef hid_t dxpl_id
+            cdef hid_t space_id = 0
+            cdef hsize_t *offset = NULL
+            cdef size_t data_size
+            cdef int rank
+            cdef uint32_t filters = 0
+            cdef hsize_t read_chunk_nbytes
+            cdef array.array data = array.array('B')
+            cdef herr_t status
+
+            if filter_mask is None:
+                # Skip all the filters
+                filters = 0xFFFF
+            else:
+                filters = int(filter_mask)
+
+            dset_id = self.id
+            dxpl_id = pdefault(dxpl)
+            space_id = H5Dget_space(self.id)
+            rank = H5Sget_simple_extent_ndims(space_id)
+
+            if offsets is not None:
+                if len(offsets) != rank:
+                    raise TypeError("offset length (%d) must match dataset rank (%d)" % (len(offsets), rank))
+
+            try:
+                offset = <hsize_t*>emalloc(sizeof(hsize_t)*rank)
+                convert_tuple(offsets, offset, rank)
+                status = H5Dget_chunk_storage_size(dset_id, offset, &read_chunk_nbytes)
+                if status < 0:
+                    raise TypeError("Error while reaching chunk storage size")
+                array.resize(data, read_chunk_nbytes)
+
+                IF HDF5_VERSION >= (1, 10, 3):
+                    status = H5Dread_chunk(dset_id, dxpl_id, offset, &filters, data.data.as_voidptr)
+                ELSE:
+                    status = H5DOread_chunk(dset_id, dxpl_id, offset, &filters, data.data.as_voidptr)
+
+                if status < 0:
+                    raise TypeError("Error while reading chunk %s", offsets)
+            finally:
+                efree(offset)
+                if space_id:
+                    H5Sclose(space_id)
+
+            return bytes(data), filters
```

Comment:
Should this match the order of the parameters in the HDF5 API? i.e. filters first, then data?

Reply:
Make sense, but the reader and writer do not use the same order. I follow the other one.
Unfortunately it will not be convenient to use, anyway.
Let me know and i update this PR.

Reply:
All the functions I could find with multiple out parameters have them in the same order for the return value. But I couldn't immediately see any others with parameters that serve for both input and output, so I don't know if there's a reason to make them an exception to the rule.